File: vtk2vtu.cpp
#include <iostream>
#include <vector>
#include <string>
#include <filesystem>

#include <vtk-9.2/vtkUnstructuredGrid.h>
#include <vtk-9.2/vtkUnstructuredGridReader.h>

namespace fs = std::filesystem;

int main(int argc, char** argv)
{
  std::vector<std::string> args(argv, argv + argc);
  fs::path path = args[1];
  if(!exists(path)) 
  {
    std::cout << "no such file or directory: "  << path << std::endl;
      std::exit(1);
  }
  auto reader = vtkUnstructuredGridReader();
}

File: vtk2vtu.py
import vtk
import sys
from concurrent.futures import ThreadPoolExecutor
import os

if(len(sys.argv) == 1):
    print("Usage: python3 VTKascii2binary {filename}")
    sys.exit(1)

def write(filename):
	reader = vtk.vtkUnstructuredGridReader()
	reader.SetFileName(filename)
	reader.ReadAllScalarsOn()
	reader.ReadAllVectorsOn()
	reader.ReadAllTensorsOn()
	reader.Update()
	writer = vtk.vtkXMLUnstructuredGridWriter()
	writer.SetFileName("{}.vtu".format(filename))
	writer.SetInputData(reader.GetOutput())
	writer.Write()
	print(" {} > {}.vtu".format(filename, filename))
	
	


with ThreadPoolExecutor(max_workers=6, thread_name_prefix="thread") as executor:
	for filename in sys.argv[1:]:
		executor.submit(write, filename)